Kickstarter-iOS/Views/Controllers/CancelPledgeViewController.swift

```diff
@@ -14,20 +33,190 @@ final class CancelPledgeViewController: UIViewController {
+
+    _ = self.cancelButton
+      |> redButtonStyle
+      |> UIButton.lens.title(for: .normal) %~ { _ in Strings.Yes_cancel_it() }
+
+    _ = self.goBackButton
+      |> greyButtonStyle
+      |> UIButton.lens.title(for: .normal) %~ { _ in Strings.No_go_back() }
   }
 
   override func bindViewModel() {
     super.bindViewModel()
+
+    self.viewModel.outputs.cancellationDetailsTextLabelValue
+      .observeForUI()
+      .observeValues { [weak self] amount, projectName in
+        self?.setCancellationDetailsAttributedText(with: amount, projectName: projectName)
+      }
+
+    self.viewModel.outputs.popCancelPledgeViewController
+      .observeForControllerAction()
+      .observeValues { [weak self] in
+        self?.navigationController?.popViewController(animated: true)
+      }
+
+    Keyboard.change
+      .observeForUI()
+      .observeValues { [weak self] change in
+        guard let self = self else { return }
+
+        self.scrollView.handleKeyboardVisibilityDidChange(change, insets: self.scrollView.contentInset)
+      }
+  }
+
+  // MARK: - Functions
+
+  private func setupConstraints() {
+    NSLayoutConstraint.activate([
+      self.rootStackView.widthAnchor.constraint(equalTo: self.view.widthAnchor),
+      self.cancelButton.heightAnchor.constraint(equalToConstant: Styles.minTouchSize.height),
+      self.goBackButton.heightAnchor.constraint(equalToConstant: Styles.minTouchSize.height),
+      self.cancellationReasonTextField.heightAnchor
+        .constraint(greaterThanOrEqualTo: self.cancelButton.heightAnchor)
+    ])
+  }
+
+  private func setCancellationDetailsAttributedText(with amount: String, projectName: String) {
+    let fullString = Strings
+      .Are_you_sure_you_wish_to_cancel_your_amount_pledge_to_project_name(
+        amount: amount,
+        project_name: projectName
+      )
+    let attributedString: NSMutableAttributedString = NSMutableAttributedString.init(string: fullString)
+    let regularFontAttribute = [NSAttributedString.Key.font: UIFont.ksr_callout()]
+    let boldFontAttribute = [NSAttributedString.Key.font: UIFont.ksr_callout().bolded]
+    let fullRange = (fullString as NSString).localizedStandardRange(of: fullString)
+    let rangeAmount: NSRange = (fullString as NSString).localizedStandardRange(of: amount)
+    let rangeProjectName: NSRange = (fullString as NSString).localizedStandardRange(of: projectName)
+
+    attributedString.addAttributes(regularFontAttribute, range: fullRange)
+    attributedString.addAttributes(boldFontAttribute, range: rangeAmount)
+    attributedString.addAttributes(boldFontAttribute, range: rangeProjectName)
+
+    _ = self.cancellationDetailsTextLabel
+      |> \.attributedText .~ attributedString
+  }
+
+  private func contentInsetsFor(traitCollection: UITraitCollection) -> UIEdgeInsets {
+    return traitCollection.preferredContentSizeCategory.isAccessibilityCategory
+      ? .init(top: 0) : .init(top: self.view.frame.size.height / 4)
```

Comment:
Doesn't `.init(top: 0)` equal `.zero`?

Also you can simply use the getter of the frame `self.view.frame.height`